transformer_lens/HookedTransformer.py

```diff
@@ -84,15 +87,16 @@ def __init__(
         self.cfg = cfg
 
         if tokenizer is not None:
-            self.set_tokenizer(tokenizer)
+            self.set_tokenizer(tokenizer, default_padding_side=default_padding_side)
```

Comment:
What is the reason for overwriting the tokenizer padding direction like this? This caused some existing trained (left-padded) models to break in a very confusing way, since the tokenizer property we were setting was being overridden without us noticing.

(in reference to #344 )

Reply:
The reason was that left-padded models were technically not supported previous to this commit, which means that only right-padded models were supported as default. Left-padded models would not have worked correctly previous to this PR.

Could you give me more details on what model you used and how the model was breaking?

Reply:
We didn't seem to have problems training and running left-padded models, but perhaps there were underlying issues which we did not notice. 

I managed to fix the issue by modifying the tokenizer in our model's `__init__` after it gets modified by the new `HookedTransformer.__init__`
https://github.com/understanding-search/maze-transformer/pull/195